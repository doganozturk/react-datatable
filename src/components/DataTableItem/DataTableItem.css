.DataTableItem td {
  padding: 15px;
  text-align: left;
  vertical-align: middle;
  font-weight: 300;
  font-size: 12px;
  color: #fff;
  border-bottom: solid 1px rgba(255, 255, 255, 0.1);
}
