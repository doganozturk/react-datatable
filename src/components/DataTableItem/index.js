import React, { Component } from 'react';

class DataTableItem extends Component {
  constructor(props) {
    super(props);

    this.state = {
      visible: false
    };

    this.toggle = this.toggle.bind(this);
    this.deleteItem = this.deleteItem.bind(this);
  }

  toggle() {
    this.setState({ visible: !this.state.visible });
  }

  deleteItem(id) {
    this.props.deleteItem(id);
  }

  render() {
    const item = this.props.item;

    let children;

    if (item.children.length) {
      children = item.children.map(child => {
        return (
          <DataTableItem
            key={child.ID}
            id={child.ID}
            item={child}
            deleteItem={this.deleteItem}
          />
        );
      });
    }

    return (
      <ul className="data-table-item">
        <li>
          {children &&
            <span className="data-table-item__toggle" onClick={this.toggle}>
              {this.state.visible ? '-' : '+'}
            </span>}
          <span>
            {item.ID}
          </span>
          <span>
            {item.Phone}
          </span>
          <span>
            {item.City}
          </span>
          <span>
            {item.Name}
          </span>
          <span
            className="data-table-item__remove"
            onClick={() => {
              this.deleteItem(item.ID);
            }}
          >
            [ delete ]
          </span>
          {children &&
            <div
              className="data-table-item__children"
              style={
                this.state.visible ? { display: 'block' } : { display: 'none' }
              }
            >
              {children}
            </div>}
        </li>
      </ul>
    );
  }
}

export default DataTableItem;
