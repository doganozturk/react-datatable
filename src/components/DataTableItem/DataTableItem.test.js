import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import DataTableItem from './';

describe(DataTableItem, () => {
  const mockItem = {
    ID: 72,
    Phone: '(134) 258-4612',
    City: 'Kukatpalle',
    Name: 'Morgan',
    children: [
      {
        ID: 82,
        Phone: '(770) 741-5391',
        City: 'San Giuliano di Puglia',
        Name: 'Burke',
        children: []
      }
    ]
  };
  const mockDeleteItem = jest.fn();
  const component = shallow(
    <DataTableItem
      id={mockItem.ID}
      item={mockItem}
      deleteItem={mockDeleteItem}
    />
  );

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <DataTableItem
        id={mockItem.ID}
        item={mockItem}
        deleteItem={mockDeleteItem}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls the passed in deleteItem function when the delete button is clicked', () => {
    component.find('span.data-table-item__remove').simulate('click');
    expect(mockDeleteItem).toBeCalled();
  });

  it('toggles the visibility of the item when toggle button is clicked', () => {
    component.find('span.data-table-item__toggle').simulate('click');
    expect(component.state('visible')).toEqual(true);
    component.find('span.data-table-item__toggle').simulate('click');
    expect(component.state('visible')).toEqual(false);
  });
});
