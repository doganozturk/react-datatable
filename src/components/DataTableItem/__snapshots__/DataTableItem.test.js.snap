// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class DataTableItem extends _react.Component {
  constructor(props) {
    super(props);

    this.state = {
      visible: false };


    this.toggle = this.toggle.bind(this);
    this.deleteItem = this.deleteItem.bind(this);
  }

  toggle() {
    this.setState({ visible: !this.state.visible });
  }

  deleteItem(id) {
    this.props.deleteItem(id);
  }

  render() {
    const item = this.props.item;

    let children;

    if (item.children.length) {
      children = item.children.map(child => {
        return (
          _react2.default.createElement(DataTableItem, {
            key: child.ID,
            id: child.ID,
            item: child,
            deleteItem: this.deleteItem, __source: { fileName: _jsxFileName, lineNumber: 31 }, __self: this }));


      });
    }

    return (
      _react2.default.createElement("ul", { className: "data-table-item", __source: { fileName: _jsxFileName, lineNumber: 42 }, __self: this },
        _react2.default.createElement("li", { __source: { fileName: _jsxFileName, lineNumber: 43 }, __self: this },
          children &&
          _react2.default.createElement("span", { className: "data-table-item__toggle", onClick: this.toggle, __source: { fileName: _jsxFileName, lineNumber: 45 }, __self: this },
            this.state.visible ? '-' : '+'),

          _react2.default.createElement("span", { __source: { fileName: _jsxFileName, lineNumber: 48 }, __self: this },
            item.ID),

          _react2.default.createElement("span", { __source: { fileName: _jsxFileName, lineNumber: 51 }, __self: this },
            item.Phone),

          _react2.default.createElement("span", { __source: { fileName: _jsxFileName, lineNumber: 54 }, __self: this },
            item.City),

          _react2.default.createElement("span", { __source: { fileName: _jsxFileName, lineNumber: 57 }, __self: this },
            item.Name),

          _react2.default.createElement("span", {
              className: "data-table-item__remove",
              onClick: () => {
                this.deleteItem(item.ID);
              }, __source: { fileName: _jsxFileName, lineNumber: 60 }, __self: this }, "[ delete ]"),



          children &&
          _react2.default.createElement("div", {
              className: "data-table-item__children",
              style:
              this.state.visible ? { display: 'block' } : { display: 'none' }, __source: { fileName: _jsxFileName, lineNumber: 69 }, __self: this },


            children))));




  }} renders and matches our snapshot 1`] = `
<ul
  className="data-table-item"
>
  <li>
    <span
      className="data-table-item__toggle"
      onClick={[Function]}
    >
      +
    </span>
    <span>
      72
    </span>
    <span>
      (134) 258-4612
    </span>
    <span>
      Kukatpalle
    </span>
    <span>
      Morgan
    </span>
    <span
      className="data-table-item__remove"
      onClick={[Function]}
    >
      [ delete ]
    </span>
    <div
      className="data-table-item__children"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <ul
        className="data-table-item"
      >
        <li>
          <span>
            82
          </span>
          <span>
            (770) 741-5391
          </span>
          <span>
            San Giuliano di Puglia
          </span>
          <span>
            Burke
          </span>
          <span
            className="data-table-item__remove"
            onClick={[Function]}
          >
            [ delete ]
          </span>
        </li>
      </ul>
    </div>
  </li>
</ul>
`;
